### ISSUE TITLE
Subclasses of external classes confusing mypy

### ISSUE THREAD

**Issue by alanhdu** [python/mypy#5445]
Using `mypy==0.620` and `python 3.6.6`, when I run `mypy --ignore-missing-imports test.py` on the following file:

```python
from PyQt5 import QtCore

class DashboardB(QtCore.QObject):
    pass
class DashboardA(QtCore.QObject):
    pass

def f() -> DashboardA:
    return DashboardB()
```

`mypy` does not raise any errors. 

 Presumably, this is happening because `QtCore.QObject` is an `Any` type to `mypy`. I assume the logic looks something like "`DashboardB` inherits from an `Any` type, which could be `DashboardA`, so `DashboardB` *could* be a subtype of `DashboardA`. I'm not sure how to actually implement this, but I'd ideally like `mypy` to know that "`DashboardB` inherits from `Any - {DashboardA}`, so `DashboardB` is not a subtype of `DashboardA`.

**Comment by ilevkivskyi** [comment]
Hm, this actually looks like a hard question. The only solution I can see is using some fake base classes and `if typing.TYPE_CHECKING: ...`, but this is apparently far from ideal solution.

**Comment by alanhdu** [comment]
Do you mean something like:

```python
if typing.TYPE_CHECKING:
    class QObject: pass
else:
    QObject = QtCore.QObject

class DashboardA(QObject): ...
class DashboardB(QObject): ...
```

I think that could work, although it'd require us to write all the methods on `QObject` that subclasses use (which seems almost equivalent to just writing type stubs at htat point :laughing:).

**Comment by ilevkivskyi** [comment]
@alanhdu Yes, this is what I meant. You can add a `__getattr__` that returns `Any` to the fake classes to suppress mypy complains. But, as I mentioned above, this is still not ideal. Maybe actually writing stubs is not such a bad idea? Now mypy supports incomplete stubs (just add a `__getattr__` wherever you want mypy to stay silent, this now works for classes, modules, and packages). You can then gradually improve the stubs without getting false negatives and share your stubs with others (using PEP 561 that is also fully supported by mypy).

**Comment by hauntsaninja** [comment]
I think this is working as expected. I don't see a good solution here and there is a not terrible workaround
